File: Vorbereitung/Übungen/2/2-3.cs
namespace Uebung;

class Program
{
    static void Main(string[] args)
    {
        var a = List(3);
        var b = List(5);

        for (int i = 1; i < 31; i++)
        {
            if (a.Contains(i))
            {
                if (b.Contains(i))
                {
                    Console.WriteLine($"FizzBuzz {i}");
                }
                else
                {
                    Console.WriteLine($"Fizz {i}");
                }
            }
            else if (b.Contains(i)) {
                Console.WriteLine($"Buzz  {i}");
            }
            else
            {
                Console.WriteLine(i);
            }
        }
    }
    static int[] List(int x)
    {
        List<int> y = new List<int>();

        for (int i = 1; i * x < 31; i++)
        {
            y.Add(i * x);
        }

        return y.ToArray();
    }
}

File: Vorbereitung/Übungen/2/2-3.py

x = 0
y = 0
matches3 = []
matches5 = []
while x < 30:
    y += 1
    x = 5 * y
    matches5.append(x)
    x = 3 * y
    matches3.append(x)
    

for i in range(1, 31, 1):
    if i in matches3:
        if i in matches5:
            print(f"Fizzbuzz {i}")
        else: 
            print(f"Fizz {i}")
    elif i in matches5:
        print(f"Buzz {i}")
    else:
        print(i)